pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/Consumer.java

```diff
@@ -66,14 +66,19 @@
     private final ConcurrentOpenHashSet<PositionImpl> pendingAcks;
 
     private final ConsumerStats stats;
+    
+    private final int maxUnackedMessages;
```

Comment:
Alright. I have added method ```shouldBlockConsumerOnUnackMsgs()``` which will enable this feature on sharedSubscription if ```maxUnackedMessages``` is > 0.